### ISSUE TITLE
why not merge this PR #517 ?

### ISSUE THREAD

**Issue by shenX-2021** [socketio/socket.io-client-java#664]
ack timeout is a very useful function. Are there plans to merge this PR #517 ?

**Comment by trueangle** [comment]
same question here

**Comment by darrachequesne** [comment]
Done!

```java
socket.emit("hello", "world", new AckWithTimeout(5000) {
    @Override
    public void onTimeout() {
        // ...
    }

    @Override
    public void onSuccess(Object... args) {
        // ...
    }
});
```

Included in version `2.1.0`.
